feat: Retain current pool management logic, but be a **LOT** less aggressive when closing idle **authenticated** connections

gkc opened the pull request:
**- What I did**
feat: Retain current inbound pool management logic, but be a **LOT** less aggressive when closing idle **authenticated** inbound connections

**- How I did it**
- Added new atServer config for 'authenticatedInboundIdleTime', defaulting to 30 days. Why 30 days? Well, there's a bug in at_lookup (there's [a PR in at_libraries](https://github.com/atsign-foundation/at_libraries/pull/354) to fix that) which leads to unhandled exceptions when server destroys idle connections. This change will ensure that that problem no longer affects existing clients
- Left existing `inboundIdleTime` at its current default value of 10 minutes, but we are now using it as the configured value for **unauthenticated** connections only
- Set default limit on total number of inbound and outbound connections to 200
- Added better instance variable comments to AtSecondaryContext
- Modified existing inbound connection pool management test accordingly. Note that the test remains essentially unchanged. Added plenty of commentary to explain the test behaviour in more detail
- Sprinkled some more comments throughout

**- How to verify it**
Tests pass

**- Description for the changelog**
feat: Retain current inbound pool management logic, but be a **LOT** less aggressive when closing idle **authenticated** inbound connections

gkc committed:
feat: Retain current pool management logic, but be a **LOT** less aggressive when closing idle **authenticated** connections
- Added new atServer config for 'authenticatedInboundIdleTimeMillis', defaulting to 30 days.
- Left existing `inboundIdleTime` at default value of 10 minutes but using it now as the configured value for **unauthenticated** connections only
- Set default limit on total number of inbound and outbound connections to 200
- Added better instance variable comments to AtSecondaryContext
- Modified existing inbound connection pool management test accordingly. Note that the test remains essentially unchanged. Added plenty of commentary to explain the test behaviour in more detail
- Sprinkled some more comments throughout

gkc committed:
style: Fix a lint warning, fix a typo

gkc committed:
build: update CHANGELOG

cconstab reviewed (dismissed):
All looks good...

My only thought was to document how we roll back to normal while its fresh in your mind now. So when the time comes its easy, for some one else to do. 
Plus set up a remind ourselves to roll back in the calendar or something so things do get migrated back when the clients are all updated..

cpswan commented on packages/at_secondary_server/config/config.yaml:
```diff
@@ -45,20 +45,31 @@ hive:
 
 # The @Protocol connection configurations.
```
Let's use the opportunity to s/@Protocol/atProtocol/

gkc replied:
👍

cpswan commented on packages/at_secondary_server/lib/src/server/at_secondary_config.dart:
```diff
@@ -61,9 +61,10 @@ class AtSecondaryConfig {
-  static final int _inboundMaxLimit = 10;
-  static final int _outboundMaxLimit = 10;
-  static final int _inboundIdleTimeMillis = 600000;
+  static final int _inboundMaxLimit = 200;
```
I know there's a bit of following suit here with the existing code, but could (should?) these be `const` rather than `static final`?

gkc replied:
> All looks good...
> 
> My only thought was to document how we roll back to normal while its fresh in your mind now. So when the time comes its easy, for some one else to do. Plus set up a remind ourselves to roll back in the calendar or something so things do get migrated back when the clients are all updated..

Sure thing. I don't see any real need to revert later but if we do then it's a change to config.yaml where I've documented these config parameters well.

I will create a ticket to review in a couple of months.

gkc replied:
> I know there's a bit of following suit here with the existing code, but could (should?) these be `const` rather than `static final`?

👍 Will change

cpswan reviewed (dismissed):
LGTM, but maybe we can make it better whilst we're here in this code?

gkc committed:
docs: replaced '@Protocol' with 'atProtocol' in config.yaml
style: replaced `static final` with `static const` where possible in at_secondary_config.dart

gkc committed:
docs: replaced '@Protocol' with 'atProtocol' in config.yaml
style: replaced `static final` with `static const` where possible in at_secondary_config.dart
